skbio/io/util.py

```diff
@@ -42,7 +47,17 @@ def _get_filehandle(filepath_or, *args, **kwargs):
     pass through.
     """
     if _is_string_or_bytes(filepath_or):
-        fh, own_fh = open(filepath_or, *args, **kwargs), True
+        if requests.compat.urlparse(filepath_or).scheme in {'http', 'https'}:
+            sess = CacheControl(requests.Session(),
+                                cache=FileCache(gettempdir()))
+            req = sess.get(filepath_or, **kwargs)
+
+            # if the response is not 200, an exception will be raised
+            req.raise_for_status()
+
+            fh, own_fh = BytesIO(req.content), True
```

Comment:
That sounds great @ElDeveloper, we should make this a first class kwarg in the registry system. (Just like `mode` is at the moment, though it will be removed soonish)